src/TextHandler.mss

```diff
@@ -111,27 +112,35 @@ function FiguredBassTextHandler (this, textObject) {
 }  //$end
 
 
-function CreateAnchoredText (this, textObj) {
+function CreateAnchoredText (this, textObject) {
     anchoredText = libmei.AnchoredText();
-    AddFormattedText(anchoredText, textObj);
+    AddFormattedText(anchoredText, textObject);
     return anchoredText;
 }  //$end
 
+function CreateDirective (this, textObject) {
+    directive = GenerateControlEvent(textObject, 'Dir');
+    AddFormattedText(directive, textObject);
+    styleIdPrefix = 'text.staff.';
+    text_style = Substring(textObject.StyleId, Length(styleIdPrefix));
```

Comment:
@rettinghaus It's clear that for what you implemented, we'll always get `@class='technique'`. I think we should document that in this form, this function is only to be registered as a text handler for `StyleId`, not for `StyleAsText` because for user defined styles, we would get not so helpful class names, like `text.staff.technique.user.0000123`. Or we can extend this a little and check if `MSplitString(o.StyleId, '.')[-2] = 'user'`, and in that case, use `textObject.StyleAsText` for the class attribute. What do you think?

Reply:
Are user defined styles a common thing? Then we definitely do the latter.

Reply:
Added a check for user defined styles.

Reply:
> Are user defined styles a common thing?

At least in our work it is.